config/src/runtime_upgrades.rs

```diff
@@ -0,0 +1,136 @@
+use std::sync::{Arc, Mutex, MutexGuard, PoisonError};
```

Comment:
May be it is a better idea to make it async? As using sync version (in async context) might lead to bugs

Reply:
I agree sync Mutex can freeze thread and tokio or any other async runtime does not account for it

Reply:
https://docs.rs/tokio/1.13.1/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use

My rationale was that in this particular case everything takes place within synchronous code. I have no problem with re-implementing using the `tokio_mutex`.

Reply:
As for `tokio::sync::Mutex`, I thought it should be avoided as much as possible in practice
https://tokio.rs/tokio/tutorial/shared-state#on-using-stdsyncmutex